cpp/arcticdb/pipeline/write_frame.cpp

```diff
@@ -31,17 +32,17 @@ namespace arcticdb::pipelines {
 using namespace arcticdb::entity;
 using namespace arcticdb::stream;
 
-std::vector<folly::Future<SliceAndKey>> write_slices(
+folly::Future<std::vector<SliceAndKey>> write_slices(
         const InputTensorFrame &frame,
-        const std::vector<FrameSlice> &slices,
+        const std::shared_ptr<std::vector<FrameSlice>> slices,
```

Comment:
`const std::shared_ptr` doesn't really make sense, generally pass by copy, or const ref if you really don't want to touch the reference counter

Reply:
agree